src/perception/se3.hpp

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <string>
+
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/transform_broadcaster.h>
+
+#include <geometry_msgs/Pose.h>
+
+class SE3 : public geometry_msgs::Pose {
+public:
+    [[nodiscard]] static SE3 fromTfTree(tf2_ros::Buffer const& buffer, std::string const& fromFrameId, std::string const& toFrameId, SE3 from, uint32_t seq);
```

Comment:
So I actually still have a few questions/concerns about this. 
- why do we need to provide a sequence number? from what I can tell in both your code and the [tf2 source code](http://docs.ros.org/en/diamondback/api/tf2_geometry_msgs/html/tf2__geometry__msgs_8h_source.html), it isn't used at all. In fact, from these forum posts [1](https://github.com/ros2/common_interfaces/issues/1#issuecomment-112621348) [2](https://discourse.ros.org/t/is-the-sequence-number-in-header-message-deprecated/1528), it seems like they are deprecated and do nothing. this also applies to `sendTfToTree`.
- I'm not sure about this being one single function, since it seems like its combining two distinct actions: getting a tf from the tf tree by providing frame ids, and then applying that tf to an SE3 object. I think these will need to be separate functions anyway since there will be times where you need to retrieve a tf from the tf tree without applying it to anything (and i dont think providing an empty SE3 is a good way to do this) and there will be times when you need to apply one SE3 object as a transform to another SE3 object. 
- I don't think this name makes sense, since to me it implies that you are creating a new SE3 object by querying the tf tree, but whats actually happening is you are transforming an existing SE3 into a different frame using the tf tree.

Reply:
Ok yeah let's get rid of sequence for sure. What do you propose we name the multiple functions you are suggesting?

Reply:
I think `fromTfTree` would be a good name for a static function that takes in two frame ids, queries the tf tree, and returns an SE3 that is the tf between those 2 frames.
then we could have a member function that takes in an SE3 and applies it as a transform to the SE3 it was called on. it could either modify the existing SE3 or return a new SE3 depending on how pure u wanna be, and ig it could be called `applyTF` or just `transform` like you had it originally. 
would also be happy to get some input on this from @niwhsa9 or anyone else

Reply:
oh wait what, why is this static if its applying a transform on an existing object?

Reply:
yes thats what I thought fromTfTree was doing when I first read through, guess I did not read carefully enough. I'm with you on that 100%

Reply:
I prose we make a shared markdown file describing all the ctors and functions of SE3

Reply:
lol idk how i missed this

Reply:
Ok the SE3 interface should be better now